src/transformers/generation_dist_process.py

```diff
@@ -0,0 +1,312 @@
+# coding=utf-8
+# Copyright 2020 HuggingFace Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+from typing import Iterable, List
+
+import numpy as np
+import torch
+from torch.nn import functional as F
+
+from .file_utils import add_start_docstrings_to_callable
+
+
+DIST_PROCESSOR_INPUTS_DOCSTRING = r"""
+    Args:
+        input_ids (:obj:`torch.LongTensor` of shape :obj:`(batch_size, sequence_length)`):
+            Indices of input sequence tokens in the vocabulary.
+
+            Indices can be obtained using :class:`~transformers.BertTokenizer`. See
+            :meth:`transformers.PreTrainedTokenizer.encode` and :meth:`transformers.PreTrainedTokenizer.__call__` for
+            details.
+
+            `What are input IDs? <../glossary.html#input-ids>`__
+        scores (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, config.vocab_size)`):
+            Prediction scores of a language modeling head. These can be scores for each vocabulary token before SoftMax
+            or scores for each vocabulary token after SoftMax.
+
+    Returns:
+        scores (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, config.vocab_size)`): The processed prediction
+        scores.
+"""
+
+
+class DistProcessorList(list):
+    """
+    This class can be used to create a list of :class:`~transformers.DistProcessor` to subsequently process a
+    :obj:`scores` input tensor. This class inherits from list and adds a specific `__call__` method to apply each
+    :class:`~transformers.DistProcessor` to the inputs.
+    """
+
+    @add_start_docstrings_to_callable(DIST_PROCESSOR_INPUTS_DOCSTRING)
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> torch.FloatTensor:
+        for processor in self:
+            scores = processor(input_ids, scores)
+        return scores
+
+
+class DistProcessor(ABC):
+    """Abstract base class for all samplers which are probability distribution warps performed during generation."""
+
+    @add_start_docstrings_to_callable(DIST_PROCESSOR_INPUTS_DOCSTRING)
```

Comment:
This method is for forward methods of PyTorch models, so you shouldn't use it in this case. `add_start_docstrings` is probably. We will rename it with @LysandreJik since you're not the first person to use it wrong ;-)

See also my comment below, if you end up never changing the docstrings, you could just define it once and for all here and not use any decorator.